tests/cypress/e2e/unit_tests/p1_fleet.spec.ts

```diff
@@ -383,73 +383,77 @@ if (/\/2\.9/.test(Cypress.env('rancher_version'))) {
-            cy.get('.col-link-detail').contains(resourceName).should('be.visible');
-            cy.open3dotsMenu(resourceName, 'Edit Config');
-
-            if (resourceType === 'ConfigMaps') {
-              cy.clickButton('Add');
-              cy.get('[data-testid="input-kv-item-key-1"]').eq(0).focus().type('test_key');
-              cy.get('div.code-mirror.as-text-area').eq(1).click().type("test_data_value");
-              cy.clickButton('Add');
-            }
-            else if (resourceType === 'Services') {
-              cy.get("input[type=number]").clear().type("6341");
-            }
-
-            else  {
-              throw new Error(`Resource "${resourceType}" is invalid  / not implemented yet`);
-            }
-
-            cy.wait(500);
-            cy.clickButton('Save');
-            cy.filterInSearchBox(resourceName);
-            cy.verifyTableRow(0, resourceName, dataToAssert);
-            // Any mutable resource will reconcile to it's original state immediately
-            // But with ConfigMaps and Services it is not because they are immutable i.e.
-            // They didn't reconciled when `correctDrift` is used.
-            cy.deleteAllFleetRepos();
+          // Add GitRepo by enabling 'correctDrift'
+          cy.fleetNamespaceToggle('fleet-default')
+          cy.addFleetGitRepo({ repoName, repoUrl, branch, path, correctDrift: 'yes' });
+          cy.clickButton('Create');
+          cy.checkGitRepoStatus(repoName, '2 / 2');
+          cy.accesMenuSelection(dsClusterName, resourceLocation, resourceType);
+          cy.nameSpaceMenuToggle(resourceNamespace);
+          cy.filterInSearchBox(resourceName);
+          cy.get('.col-link-detail').contains(resourceName).should('be.visible');
+          cy.open3dotsMenu(resourceName, 'Edit Config');
+
+          if (resourceType === 'ConfigMaps') {
+            cy.clickButton('Add');
+            cy.get('[data-testid="input-kv-item-key-1"]').eq(0).focus().type('test_key');
+            cy.get('div.code-mirror.as-text-area').eq(1).click().type("test_data_value");
+            cy.clickButton('Add');
+          }
+          else if (resourceType === 'Services') {
+            cy.get("input[type=number]").clear().type("6341");
+          }
+
+          else  {
+            throw new Error(`Resource "${resourceType}" is invalid  / not implemented yet`);
+          }
 
-            // Delete leftover resources if there are any.
-            cy.accesMenuSelection(dsClusterName, resourceLocation, resourceType);
+          cy.wait(500);
+          cy.clickButton('Save');
+          cy.filterInSearchBox(resourceName);
+          cy.verifyTableRow(0, resourceName, dataToAssert);
+          // Any mutable resource will reconcile to it's original state immediately
+          // But with ConfigMaps and Services it is not because they are immutable i.e.
+          // They didn't reconciled when `correctDrift` is used.
+          cy.deleteAllFleetRepos();
```

Comment:
Before deleting the repos let's add 2 checks:

- Check 1: 

Ensure Gitrepo does is not in Error state as here: (sought in 2.8)
![image (1)](https://github.com/rancher/fleet-e2e/assets/37271841/59540d62-86c8-492f-ba88-9a4b268f4682)
I guess a `cy.verifyTableRow(0, Active, 1/1);` should be ok

- Check 2

Lets add an extra step to check alll ds clusters are ok to avoid silent errors (sought in 2.9)
 ![image (2)](https://github.com/rancher/fleet-e2e/assets/37271841/7aeac671-187a-407a-8e23-efe4738876f6)

Reply:
Added logic to check `GitRepo` status and `Cluster` status after all `correctDrift` operations performed.